src/main/java/tlylt/haha/Storage.java

```diff
@@ -0,0 +1,47 @@
+package tlylt.haha;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Utility for storing tasks to file.
+ */
+public class Storage {
+    private Path file;
+
+    /**
+     * Initialises by checking and creating storage file if needed.
+     */
+    public Storage() {
+        try {
+            String currentDir = System.getProperty("user.dir");
+            Path folder = Paths.get(currentDir, "Haha_data");
```

Comment:
Perhaps specify filePath and folderPath instead? Otherwise it might sound misleading.